mdoc/src/main/scala-3/mdoc/internal/markdown/MarkdownCompiler.scala

```diff
@@ -148,41 +163,77 @@ class MarkdownCompiler(
     }
   }
 
-  private def toMetaPosition(
-      edit: TokenEditDistance,
-      position: SourcePosition
-  ): Position = {
+  def fail(edit: TokenEditDistance, input: Input, sectionPos: Position): String = {
+    val compiler = new Compiler
+    val freshContext = context.fresh
+
+    val run = compiler.newRun(using freshContext)
+
+    val inputs = List(input).map(toSource)
+
+    run.compileSources(inputs)
+    val runContext = run.runContext
+    val out = new ByteArrayOutputStream()
+    val ps = new PrintStream(out)
+    
+    runContext.reporter.pendingMessages(using context).foreach { diagnostic =>
+      val msg = nullableMessage(diagnostic.message)
+      val mpos = toMetaPosition(edit, diagnostic.position.get)
+      if(sectionPos.contains(mpos) || diagnostic.level == IDiagnostic.ERROR) {
+        val severity = diagnostic.level match {
+          case IDiagnostic.ERROR => "error"
+          case IDiagnostic.WARNING => "warn"
+          case IDiagnostic.INFO => "info"
+        }
+        val formatted = 
+          PositionSyntax.formatMessage(mpos, severity, "\n" + msg, includePath = false)
+        ps.println(formatted)
+      }
+    }
+    
+    out.toString()
+  }
+
+
+  def toMetaPosition(edit: TokenEditDistance, pos: SourcePosition): Position = {
     def toOffsetPosition(offset: Int): Position = {
       edit.toOriginal(offset) match {
-        case Left(_) =>
-          Position.None
-        case Right(p) =>
-          p.toUnslicedPosition
+        case Left(_) => Position.None
+        case Right(p) => p.toUnslicedPosition
       }
     }
-    (edit.toOriginal(position.start), edit.toOriginal(position.end - 1)) match {
+
+    val start = pos.start
+    val end = pos.end
+
+    (edit.toOriginal(start), edit.toOriginal(end - 1)) match {
       case (Right(start), Right(end)) =>
-        Position.Range(start.input, start.start, end.end).toUnslicedPosition
+          Position.Range(start.input, start.start, end.end).toUnslicedPosition
```

Comment:
Basically, to make that one Worksheets test pass, I need to do 

```scala
Position.Range(start.input, start.start - 2, end.end).toUnslicedPosition
```

(and you can guess that -2 is arbitrary)

But this then breaks several other tests:

```scala
=> Diff (- obtained, + expected)
-error: yrangepos-scala3.md:3:9:
+error: yrangepos-scala3.md:3:11:
 Found:    String
 List[Int]("".length.toString)
-        ^^^^^^^^^^^^^^^^^^^^
+          ^^^^^^^^^^^^^^^^^^
```

Which is expected, given how arbitrary `2` is. You [PR](https://github.com/scalameta/mdoc/pull/386) mentions that that workaround might have issues - feels like after running all the tests on dotty something else is required.

I can keep looking for a workaround to satisfy all tests, unless you have a better suggestion?

Reply:
Looking at what comes from the compiler + what's in the TokenEditDistance:

```
val [115..118) <-> val [617..620)
  [118..119) <->   [620..621)
xx [119..121) <-> xx [621..623)
  [121..122) <->   [623..624)
= [122..123) <-> = [624..625)
  [123..124) <->   [625..626)
fn [124..126) <-> fn [626..628)
\n [126..127) <-> \n [628..629)
EOF [127..127) <-> EOF [721..721)
()

 Original: dotty-ambiguous-implicit.scala:<628..628>. 
 Edited Start: 126:127, 
 Edited End: 124:126 
```

This:

```
fn [124..126) <-> fn [626..628)
```

Looks like a bug.. somewhere?

Reply:
I remember that this was not pretty, I think we can leave it as is and I can take a look in another PR? Just maybe let's track that under an issue.

Reply:
> I think we can leave it as is

Reply:
I've changed the test to what it output now.